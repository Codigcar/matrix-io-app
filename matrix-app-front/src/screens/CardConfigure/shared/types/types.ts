/*
    TODO: This type `PropsWithTestID` will be moved to a global types (shared/types)
    when we have all changes of Dambert related to refactor
    Mod. Date: 14/12/2023
*/
export type PropsWithTestID<P> = P & { testID?: string | undefined };

export type Card = {
  id: string;
  label: string;
  type: string;
  status: boolean;
  isBlock: boolean;
  loading: boolean;
  requireChangePin?: boolean;
};

export type Restriction = {
  id: number;
  label: string;
  type: string;
  restriction: string;
  status: boolean;
  loading: boolean;
  isHiding?: boolean;
};
