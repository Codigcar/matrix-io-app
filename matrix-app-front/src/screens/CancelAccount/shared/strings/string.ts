import { i18n } from 'src/utils/core/MTXStrings';

export const string = {
  // TODO: Move these to shared/strings when we have all changes of Dambert related to refactor
  appName: i18n.t('app-name'),
  // TODO END
  cancelAccountCompleteSecondSectionText1: i18n.t(
    'cancelAccount:cancelAccountComplete.second-section-text-1',
  ),
  cancelAccountCompleteSecondSectionText3: i18n.t(
    'cancelAccount:cancelAccountComplete.second-section-text-3',
  ),
  cancelAccountCompleteSecondSectionText2: i18n.t(
    'cancelAccount:cancelAccountComplete.second-section-text-2',
  ),
  cancelAccountCompleteFirstSectionTitle: i18n.t(
    'cancelAccount:cancelAccountComplete.first-section-title',
  ),
  cancelAccountCompleteSecondSectionTitle: i18n.t(
    'cancelAccount:cancelAccountComplete.second-section-title',
  ),
  cancelAccountCompleteThirdSectionText1: i18n.t(
    'cancelAccount:cancelAccountComplete.third-section-text-1',
  ),
  cancelAccountCompleteThirdSectionText2: i18n.t(
    'cancelAccount:cancelAccountComplete.third-section-text-2',
  ),
  cancelAccountCompleteThirdSectionText3: i18n.t(
    'cancelAccount:cancelAccountComplete.third-section-text-3',
  ),
  cancelAccountCompleteHeader: i18n.t('cancelAccount:cancelAccountComplete.header'),
  cancelAccountCompleteTitle: i18n.t('cancelAccount:cancelAccountComplete.title'),
  cancelAccountCompleteSubtitle: i18n.t('cancelAccount:cancelAccountComplete.subtitle'),
  cancelAccountCompleteChatButton: i18n.t('cancelAccount:cancelAccountComplete.chat-button'),
  cancelAccontCompletePaymentButton: i18n.t('cancelAccount:cancelAccountComplete.payment-button'),
  cancelAccountCompleteContinueButton: i18n.t(
    'cancelAccount:cancelAccountComplete.continue-button',
  ),
  cancelAccountSurveyModalTitle: i18n.t('cancelAccount:cancelAccount.survey.modal.title'),
  cancelAccountSurveyModalSubtitle1: i18n.t('cancelAccount:cancelAccount.survey.modal.subtitle-1'),
  cancelAccountSurveyModalSubtitle2: i18n.t('cancelAccount:cancelAccount.survey.modal.subtitle-2'),
  cancelAccountSurveyModalTextTitle: i18n.t('cancelAccount:cancelAccount.survey.modal.text-title'),
  cancelAccountSurveyModalTextSubtitle: i18n.t(
    'cancelAccount:cancelAccount.survey.modal.text-subtitle',
  ),
  cancelAccountSurveyModalCheckboxInfo: i18n.t(
    'cancelAccount:cancelAccount.survey.modal.checkbox-info',
  ),
  cancelAccountSurveyModalBtnCancelAccount: i18n.t(
    'cancelAccount:cancelAccount.survey.modal.btnCancelAccount',
  ),
  cancelAccountHeader: i18n.t('cancelAccount:cancelAccount.header'),
  cancelAccountSurveyTitle: i18n.t('cancelAccount:cancelAccount.survey.title'),
  cancelAccountSurveySubtitle: i18n.t('cancelAccount:cancelAccount.survey.subtitle'),
  cancelAccountSurveyQuestion: i18n.t('cancelAccount:cancelAccount.survey.question'),
  cancelAccountSurveyOption4Title: i18n.t('cancelAccount:cancelAccount.survey.option-4-title'),
  cancelAccountSurveyOption4Placeholder: i18n.t(
    'cancelAccount:cancelAccount.survey.option-4-placeholder',
  ),
  cancelAccountSurveyContinueButton: i18n.t('cancelAccount:cancelAccount.survey.continue-button'),
  cancelAccountSurveyOption1: i18n.t('cancelAccount:cancelAccount.survey.option-1'),
  cancelAccountSurveyOption2: i18n.t('cancelAccount:cancelAccount.survey.option-2'),
  cancelAccountSurveyOption3: i18n.t('cancelAccount:cancelAccount.survey.option-3'),
  cancelAccountSurveyOption4: i18n.t('cancelAccount:cancelAccount.survey.option-4'),
  cancelAccountTitle: i18n.t('cancelAccount:cancelAccount.title'),
  cancelAccountSubtitle1: i18n.t('cancelAccount:cancelAccount.subtitle1'),
  cancelAccountSubtitle2: i18n.t('cancelAccount:cancelAccount.subtitle2'),
  cancelAccountQuestionLabel: i18n.t('cancelAccount:cancelAccount.question-label'),
  cancelAccountContinueButton: i18n.t('cancelAccount:cancelAccount.continue-button'),
  cancelAccountWaitingErrorTitle: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.errorTitle',
  ),
  cancelAccountWaitingErrorSubtitle: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.errorSubtitle',
  ),
  cancelAccountWaitingErrorDescription: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.errorDescription',
  ),
  cancelAccountWaitingErrorButtonLabel: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.errorButtonLabel',
  ),
  cancelAccountWaitingWaitingTitle: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.waiting-title',
  ),
  cancelAccountWaitingWaitingSubtitle1: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.waiting-subtitle-1',
  ),
  cancelAccountWaitingWaitingSubtitle2: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.waiting-subtitle-2',
  ),
};
