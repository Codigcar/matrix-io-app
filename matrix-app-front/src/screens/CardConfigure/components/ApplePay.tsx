import React from 'react';
// Assets
import { i18n } from 'src/utils/core/MTXStrings';
// Components
import { Text, fonts, Box } from 'matrix-ui-components';
import AppleRectangle from 'assets/svgs/apple_black.svg';
import PayWallet from 'assets/svgs/pay_wallet.svg';
import { Platform, Pressable } from 'react-native';

interface IApplePayProps {
  onPress: () => void;
}

const ApplePay = ({ onPress }: IApplePayProps) => (
  <Pressable onPress={onPress}>
    <Box
      backgroundColor="white"
      borderWidth={1}
      borderRadius={16}
      borderColor="gray100"
      paddingHorizontal="spacing-m"
      pt="spacing-xs"
      pb="spacing-xs"
      shadowColor="primaryMedium"
      shadowOffset={{
        width: 0,
        height: 5,
      }}
      shadowOpacity={0.25}
      shadowRadius={5}
      elevation={5}
    >
      <Box my="spacing-xxs" flexDirection="row" justifyContent="flex-start" alignItems="center">
        {Platform.OS === 'android' ? (
          <PayWallet />
        ) : (
          <AppleRectangle />
        )}
        <Box justifyContent="center" alignItems="flex-start">
          <Text
            textAlign="left"
            ml="spacing-xxs"
            mr="spacing-s"
            paddingHorizontal="spacing-xxxs"
            mt="spacing-xxxxs"
            variant="label"
            fontFamily={fonts.euclidCircularMedium}
            fontSize={14}
            lineHeight={14}
            color="primaryDarkest"
          >
            {i18n.t(Platform.OS === 'ios' ? 'apple-pay' : 'wallet:wallet-flow.pay-with-nfc')}
          </Text>
          <Text
            textAlign="left"
            my="spacing-xxxs"
            ml="spacing-xxs"
            mr="spacing-s"
            paddingHorizontal="spacing-xxxs"
            variant="label"
            fontSize={12}
            lineHeight={12}
            fontFamily={fonts.euclidCircularRegular}
            color="primaryDark"
          >
            {i18n.t('configure-card-2-detail-1')}
          </Text>
        </Box>
      </Box>
    </Box>
  </Pressable>
);

export default ApplePay;
