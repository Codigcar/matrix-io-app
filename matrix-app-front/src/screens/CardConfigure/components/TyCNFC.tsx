import React from 'react';
import { i18n } from 'src/utils/core/MTXStrings';
import { Text, fonts, Box } from 'matrix-ui-components';

const TyCNFC = () => (
  <Box
    backgroundColor="white"
    borderWidth={1}
    borderRadius={16}
    borderColor="gray100"
    p="spacing-m"
    pb="spacing-m"
    shadowColor="primaryMedium"
    shadowOffset={{
      width: 0,
      height: 5,
    }}
    shadowOpacity={0.25}
    shadowRadius={5}
    elevation={5}
  >
    <Text
      textAlign="left"
      variant="body"
      fontFamily={fonts.euclidCircularSemibold}
      fontSize={16}
      lineHeight={16}
    >
      {i18n.t('wallet-flow.pay-with_nfc_tyc_title')}
    </Text>
    <Text
      textAlign="left"
      variant="body"
      mt="spacing-m"
      fontFamily={fonts.euclidCircularLight}
      fontSize={12}
      lineHeight={16}
    >
      {i18n.t('wallet-flow.pay-with_nfc_tyc')}
    </Text>
  </Box>
);

export default TyCNFC;
