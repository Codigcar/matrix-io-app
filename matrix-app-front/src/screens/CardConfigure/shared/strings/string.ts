import { i18n } from 'src/utils/core/MTXStrings';

export const string = {
  activated: i18n.t('cardConfigure:status.activated'),
  desactivated: i18n.t('cardConfigure:status.deactivated'),
  noCard: i18n.t('cardConfigure:status.no-card'),
  cardBlocked: i18n.t('cardConfigure:status.card-blocked'),
  activatedPurchases: i18n.t('cardConfigure:status.activated-purchases'),
  desactivatedPurchases: i18n.t('cardConfigure:status.deactivated-purchases'),
  configureCardSubtitle: i18n.t('cardConfigure:home.configure-card-0-subtitle'),
  configureCardTitle: i18n.t('cardConfigure:home.configure-card-title'),
  configureCard0Title: i18n.t('cardConfigure:home.configure-card-0-title'),
  configureCard0SubtitleInit: i18n.t('cardConfigure:home.configure-card-0-subtitle-init'),
  walletFlowNFCIncopatibleTitle: i18n.t('cardConfigure:wallet-flow.nfc.incompatible.title'),
  walletFlowNFCIncopatibleMessage: i18n.t('cardConfigure:wallet-flow.nfc.incompatible.message'),
  walletFlowUnderstood: i18n.t('cardConfigure:wallet-flow.understood'),
  cardConfigurationVirtualCard: i18n.t('cardConfigure:virtual-card'),
  cardConfigurationPhysicalCard: i18n.t('cardConfigure:physical-card'),
  cardConfigurationSupplementaryCardLabel: i18n.t('cardConfigure:supplementary-card.label'),
  cardConfigurationSupplementaryCardHelper: i18n.t('cardConfigure:supplementary-card.helper'),
  cardConfigurationSupplementaryCardMessage: i18n.t('cardConfigure:supplementary-card.message'),
  cardConfigurationSupplementaryCardActionText: i18n.t(
    'cardConfigure:supplementary-card.action-text',
  ),
  configureCard0Detail0: i18n.t('cardConfigure:home.configure-card-0-detail-0'),
  configureCard1Detail0: i18n.t('cardConfigure:home.configure-card-1-detail-0'),
  configureCard0Detail1: i18n.t('cardConfigure:home.configure-card-0-detail-1'),
  configureCard1Detail1: i18n.t('cardConfigure:home.configure-card-1-detail-1'),
  configureCardError01: i18n.t('cardConfigure:home.configure-card-error-01'),
  configureCardError02: i18n.t('cardConfigure:home.configure-card-error-02'),
  configureCardInternetDesactivated: i18n.t(
    'cardConfigure:home.configure-card-internet-deactivated',
  ),
  configureCardInternetActivated: i18n.t('cardConfigure:home.configure-card-internet-activated'),
  configureCardForeignDesactivated: i18n.t('cardConfigure:home.configure-card-foreign-deactivated'),
  configureCardForeignActivated: i18n.t('cardConfigure:home.configure-card-foreign-activated'),
};
