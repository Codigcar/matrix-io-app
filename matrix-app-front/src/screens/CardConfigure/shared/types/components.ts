import { PropsWithChildren } from 'react';
import { PropsWithTestID } from './types';

export interface CardItemProps extends PropsWithChildren<any>, PropsWithTestID<any> {}

export interface IconSwitchListItemProps extends PropsWithTestID<any> {
  label: string;
  type?: string;
  status?: boolean;
  onChange: any;
  loading?: boolean;
  isBlocked?: boolean;
}

export interface SupplementaryCardChangePinProps extends PropsWithTestID<any> {
  onPress: () => void;
}
