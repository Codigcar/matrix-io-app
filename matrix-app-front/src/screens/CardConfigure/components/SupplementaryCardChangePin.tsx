import React from 'react';
import { Box, Text } from 'matrix-ui-components';
import { TouchableOpacity } from 'react-native';
import { string } from '../shared/strings/string';
import { testID } from '../shared/strings/testID';
import { SupplementaryCardChangePinProps } from '../shared/types/components';

const SupplementaryCardChangePin: React.FC<SupplementaryCardChangePinProps> = (props) => {
  const { onPress, testID: testIDProp } = props;
  return (
    <Box testID={testIDProp} mt="spacing-xs" ml="spacing-xxxxxs">
      <Box flexDirection="row" width="100%" justifyContent="space-between" mt="spacing-xxxs">
        <Text variant="Subtitle16Semibold" color="primary1000">
          {string.cardConfigurationSupplementaryCardLabel}
        </Text>
        <Text variant="body13pxRegular" color="primary500">
          {string.cardConfigurationSupplementaryCardHelper}
        </Text>
      </Box>
      <Box flexDirection="row" width="100%" mt="spacing-xxxs">
        <Text variant="body13Regular" color="primary700">
          {string.cardConfigurationSupplementaryCardMessage}
        </Text>
        <TouchableOpacity
          testID={testID.supplementaryCardChangePinActionButton}
          onPress={() => onPress()}
        >
          <Text
            variant="body13Regular"
            color="complementaryIndigo600"
            textDecorationLine="underline"
          >
            {string.cardConfigurationSupplementaryCardActionText}
          </Text>
        </TouchableOpacity>
      </Box>
    </Box>
  );
};

export default SupplementaryCardChangePin;
