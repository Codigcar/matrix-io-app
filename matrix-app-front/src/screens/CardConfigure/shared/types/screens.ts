import { NavigationPropsType } from 'src/types/types';
import { PropsWithTestID } from './types';

export interface CardConfigureHomeProps extends PropsWithTestID<any>, NavigationPropsType {}
