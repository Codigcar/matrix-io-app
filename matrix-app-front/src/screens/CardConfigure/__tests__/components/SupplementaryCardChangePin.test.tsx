import React from 'react';
import { fireEvent, render } from 'src/matrix-ui-components/utils/test-utils';
import SupplementaryCardChangePin from '../../components/SupplementaryCardChangePin';
import { testID } from '../../shared/strings/testID';
import { string } from '../../shared/strings/string';
import { SupplementaryCardChangePinProps } from '../../shared/types/components';

const mainComponentTestId = testID.supplementaryCardChangePin;

const onPressMock = jest.fn();

const componentRender = (props: SupplementaryCardChangePinProps) =>
  render(
    <SupplementaryCardChangePin
      testID={mainComponentTestId}
      onPress={props.onPress ?? jest.fn()}
    />,
  );

describe('SupplementaryCardChangePin component', () => {
  it('should render without errors', () => {
    const { getByTestId } = componentRender({ onPress: onPressMock });
    const component = getByTestId(mainComponentTestId);
    expect(component).toBeTruthy();
  });

  it('should render all labels', () => {
    const { getByText } = componentRender({ onPress: onPressMock });
    const label1 = getByText(string.cardConfigurationSupplementaryCardLabel);
    const label2 = getByText(string.cardConfigurationSupplementaryCardHelper);
    const label3 = getByText(string.cardConfigurationSupplementaryCardMessage);
    const label4 = getByText(string.cardConfigurationSupplementaryCardActionText);
    expect(label1).toBeTruthy();
    expect(label2).toBeTruthy();
    expect(label3).toBeTruthy();
    expect(label4).toBeTruthy();
  });

  it('should press button', () => {
    const onPressForButtonMock = jest.fn();
    const { getByTestId } = componentRender({ onPress: onPressForButtonMock });
    const actionButton = getByTestId(testID.supplementaryCardChangePinActionButton);
    fireEvent.press(actionButton);
    expect(onPressForButtonMock).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const { toJSON } = componentRender({ onPress: onPressMock });
    expect(toJSON()).toMatchSnapshot();
  });
});
