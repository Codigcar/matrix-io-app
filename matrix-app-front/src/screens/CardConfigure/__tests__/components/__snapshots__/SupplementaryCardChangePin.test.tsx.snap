// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplementaryCardChangePin component should match snapshot 1`] = `
<View
  style={
    [
      {
        "marginLeft": 2,
        "marginTop": 14,
      },
    ]
  }
  testID="supplementaryCardChangePinId"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 4,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Outfit-SemiBold",
            "fontSize": 14,
            "fontWeight": "600",
            "lineHeight": 16.8,
          },
        ]
      }
    >
      Tarjeta física
    </Text>
    <Text
      style={
        [
          {
            "color": "#8F8F8F",
            "fontFamily": "Outfit-Regular",
            "fontSize": 13,
            "letterSpacing": 0.16,
            "lineHeight": 18.2,
          },
        ]
      }
    >
      Activación pendiente
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginTop": 4,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#666666",
            "fontFamily": "Outfit-Regular",
            "fontSize": 11.375,
            "letterSpacing": 0.16,
            "lineHeight": 15.925,
          },
        ]
      }
    >
      Para activar tu tarjeta física 
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="supplementaryCardChangePinActionButtontId"
    >
      <Text
        style={
          [
            {
              "color": "#3C5BB9",
              "fontFamily": "Outfit-Regular",
              "fontSize": 11.375,
              "letterSpacing": 0.16,
              "lineHeight": 15.925,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        cambia el PIN
      </Text>
    </View>
  </View>
</View>
`;
